apps/web/src/services/dashboards/pages/DashboardsMainPage.vue

```diff
@@ -119,14 +129,66 @@ const queryState = reactive({
 });
 
 /* Util */
-const getSearchedTreeData = (treeData: TreeNode<DashboardTreeDataType>[], searchFilters: ConsoleFilter[]): TreeNode<DashboardTreeDataType>[] => {
-    if (!searchFilters.length) return treeData;
-    return treeData.filter((node) => {
-        if (node.data.type === 'DASHBOARD') return true;
-        return !!node.children?.length;
+const getSearchedTreeData = (treeData: TreeNode<DashboardTreeDataType>[], searchedDashboardIdList: Set<string>): TreeNode<DashboardTreeDataType>[] => {
+    if (state.isSearching) return [];
+    if (!state.searchFilters.length) return treeData;
+    const _results = [] as TreeNode<DashboardTreeDataType>[];
+    treeData.forEach((node) => {
+        const _node = cloneDeep(node);
+        if (_node.data.type === 'DASHBOARD') {
+            if (searchedDashboardIdList.has(_node.data.id)) {
+                _results.push(_node);
+            }
+        } else {
+            const _children = getSearchedTreeData(_node.children || [], searchedDashboardIdList);
+            if (_children.length) {
+                _node.children = _children;
+                _results.push(_node);
+            }
+        }
```

Comment:
suggestion: what if clone deep when it's really necessary?

```suggestion
        if (node.data.type === 'DASHBOARD') {
            if (searchedDashboardIdList.has(node.data.id)) {
                _results.push(cloneDeep(node));
            }
        } else {
            const _children = getSearchedTreeData(node.children || [], searchedDashboardIdList);
            if (_children.length) {
                const _node = cloneDeep(node);
                _node.children = _children;
                _results.push(_node);
            }
        }
```